src/handlers/serenity.rs

```diff
@@ -134,18 +134,14 @@ impl SerenityHandler {
                                 .kind(ApplicationCommandOptionType::String)
                                 .required(true)
                         })
-                })
-                .create_application_command(|command| {
-                    command
-                        .name("playtop")
-                        .description("Places a track on the top of the queue")
-                        .default_permission(false)
                         .create_option(|option| {
                             option
-                                .name("query")
-                                .description("The media to play")
+                                .name("flag")
+                                .description("Play command flag")
                                 .kind(ApplicationCommandOptionType::String)
-                                .required(true)
+                                .add_string_choice("Place this song at the top of the queue", "top")
+                                .add_string_choice("Place all songs in the playlist in the queue", "all")
+                                .required(false)
```

Comment:
We could make use of [subcommands](https://discord.com/developers/docs/interactions/application-commands#example-walkthrough) here for the flags. Because right now this is a little finicky to interact with:

![image](https://user-images.githubusercontent.com/5618493/153754506-2a78352d-ae64-44d8-83b2-f2bcca0b8780.png)

![image](https://user-images.githubusercontent.com/5618493/153754491-5fddbc54-8b33-496c-8dc4-6c6bbd0271db.png)

This grouping would be a little clearer and would allow for specific help text (e.g. **/play query top** with helper text **"Place track at the top of the queue"**):

![image](https://user-images.githubusercontent.com/5618493/153754981-85c73e08-b48f-4cdd-9504-4f4cfe08dc93.png)

Reply:
I don't know if I agree with you here @aquelemiguel, since this would pollute the command listing a bit. Additionally, the flag is optional, so you don't even need to type it.

Reply:
I like the suggested approach much more too.

Reply:
@afonsojramos A few considerations: 
* Frequently used commands always show up on top and there's autocomplete (plus we already had **/playtop** as a separate command and it was okay).
* UX-wise currently this is really awkward to interact with. Final command text looks weird and too long (if anything, it should just be `flag: top`) and there's no place for helper text.
* This is the expected use-case for subcommands.
 
I recommend pulling the PR and messing with it for a while.

Reply:
Yeah, I agree with the remark about the helper text. That alone makes me want to change the implementation, but functionally I think I still prefer the flags.

Its quick and optional, but I wonder if there's some standard already in place to leverage instead of debating this single use case.

Reply:
> Yeah, I agree with the remark about the helper text. That alone makes me want to change the implementation, but functionally I think I still prefer the flags.

Yup, I think we can all agree flags are faster for old prefix-based commands.

> Its quick and optional, but I wonder if there's some standard already in place to leverage instead of debating this single use case.

For slash commands I'm pretty certain subcommands are the standard, please check [the docs](https://discord.com/developers/docs/interactions/application-commands#subcommands-and-subcommand-groups).
@afonsojramos Discord might nest subcommands in the future:

![image](https://user-images.githubusercontent.com/5618493/153757256-8ba314b3-a07e-4f5f-98ce-2f9c9d476bc1.png)

Reply:
And flags actually seem slower to use than a specific subcommand. Flags as a concept are useful in CLI applications, for slash commands I don't see it being coherent or useful. I much prefer the subcommand approach.

Reply:
Alright, I'll start working on a subcommand implementation, but as unfamiliar with rust as I am, I am even more unfamiliar with serenity's subcommand system so I'll need another day or two. Are there any other things I should look into? Everyone happy with leveraging the PlayFlag enum?

Should we continue to let the interaction attributes override the command arguments or should we make that only work if the argument isn't explicitly given? Or should the argument that are being overridden be removed? I'm sure I'll have some input here when I go to implement it, but for future functionality is there anything that comes to mind?